File: email-service/src/main/java/com/backend/email/controller/HealthCheckController.java
package com.backend.email.controller;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/health")
public class HealthCheckController {

  @GetMapping("/check")
  @ResponseStatus(HttpStatus.OK)
  public Map<String, String> healthCheck() {
    HashMap<String, String> status = new HashMap<>();
    status.put("status", "ok");
    return status;

  }
}

File: like-service/src/main/kotlin/com/backend/like/controller/HealthCheckController.kt
package com.backend.like.controller

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/health")
class HealthCheckController {

    @GetMapping("/check")
    @ResponseStatus(HttpStatus.OK)
    fun healthcheck(): Map<String, String> {
        var map = mutableMapOf<String, String>()
        map["status"] = "ok"
        return map
    }
}